97.md

```diff
@@ -0,0 +1,25 @@
+NIP-97
+======
+
+Nostr Login
+-----------
+
+`draft` `optional`
+
+This NIP defines a method for a way for users to prove to apps that they control a certain private key. This can be used to enable login for services like the management panel of image hosts.
+
+## Login URI
+
+A login URI is defined of the format `nostr+login:<domain>:<identifier>` where `domain` MUST be a valid DNS domain or .onion service. The `identifier` MUST only consist of `A-Z a-z 0-9 _ - .`.
+
+This login URI can be presented as a clickable link, a QR code or a copyable string.
+
+## Login process
+
+A client that wishes to log in to a service SHOULD display the domain associated with the service to the end user before allowing them to log in to prevent services showing a login string for another service.
+
+After the user approves the login, the client should send a POST request to `/.well-known/nostr-login` to the `domain` with the `i` query parameter set to the `identifier` and a valid NIP-98 authentication header present.
```

Comment:
We should also specify the return payload, to indicate success or failure. Perhaps something like:

Return HTTP code:
- 200: Login successful
- 401: NIP-98 auth failed
- ...

```
{
    "status": <"success" or "error">
    "redirect_url": <OPTIONAL_URL_TO_CONTINUE_LOGIN>
}
```

This would help clients know what to display to the user after they accept the login

Reply:
On top of my comment above, I believe whatever session auth token or secret auth information needs to be (or at least should be) on this payload (Maybe embedded into the `redirect_url` for simplicity), to avoid phishing attack vectors.

To understand the possibility of a phishing vulnerability, consider the following:
1. Alice clicks on a link to `fake.com/login` (A fake website imitating `real.com`)
2. `fake.com` detects Alice entered the website, and then crawls `real.com/login` to obtain the NIP-97 link `nostr+login:real.com:abc123`. It then copies and displays this URL as a QR code on `fake.com/login`
3. Alice, unknowingly, scans the QR code which opens her Nostr client. She sees a prompt "Do you want to login into `real.com`?"
4. She clicks "confirm". The Nostr client proceeds to send the NIP-98 authenticated request to `real.com`
5. In this particular implementation of `real.com` _(which meets the specs of this NIP-97 draft)_, as soon as the `real.com` server receives the NIP-98 auth'd event, it generates an auth session token and sends it to the original browser session that received the login URL — to allow the login to proceed on the same tab.
6. Thus, in this particular implementation, the attacker successfully tricked Alice into giving the them access to her account on `real.com`.

However, if `real.com` implemented it in such a way that provides the nostr client with the redirect url containing the auth session token info, this type of phishing attack would not be possible

Reply:
> that is the user's fault and therefore this is a non-issue
> 
> your proposal breaks logging in on desktop using a mobile device also

I see, you have a point. However, it is reasonable to allow developers to optionally implement the flow with redirect urls if the tradeoff makes sense for their particular use case. 

Besides, while @jb55 and I were developing a very similar flow for our Damus Purple checkout page, we found that such redirect URL mechanism helped improve the UX on mobile-only flows (Some early testers were confused when we did not have them)

The best solution is to make the `redirect_url` an optional field on the JSON response, and thus allow each website to choose whether or not to use it (Whichever works best for their use case).